### PR TITLE
aead-stream: extract implementation from the aead crate

### PR DESCRIPTION
**Pull request by newpavlov**
This PR extracts implementation of the STREAM construction from the `aead` crate into a separate crate as was proposed in https://github.com/RustCrypto/traits/issues/1665.

### CODE REVIEW

File: aead-stream/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+[package]
+name = "aead-stream"
+version = "0.1.0"
```
Comment (tarcieri) [inline aead-stream/Cargo.toml:3]: Would suggest syncing the version with `aead` so it's easy to know which versions are compatible

```suggestion
version = "0.6.0-pre"
```

File: aead-stream/CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# Changelog
+All notable changes to this project will be documented in this file.
+
+The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
+and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
+
+## 0.1.0 (2024-xx-xx)
```
Comment (tarcieri) [inline aead-stream/CHANGELOG.md:7]: Version number also here (maybe just leave it out for now?)

File: aead-stream/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+    /// Compute the full AEAD nonce including the STREAM counter and last
+    /// block flag.
+    fn aead_nonce(&self, position: u32, last_block: bool) -> Result<aead::Nonce<A>> {
+        if position > Self::COUNTER_MAX {
+            return Err(Error);
+        }
+
+        let mut result = Array::default();
+
+        // TODO(tarcieri): use `generic_array::sequence::Concat` (or const generics)
```
Comment (tarcieri) [inline aead-stream/src/lib.rs:220]: Sidebar: should probably fix this, or at least update it to use `hybrid-array`

File: aead-stream/src/lib.rs [hunk 4]
```diff
@@ -0,0 +1,229 @@
+#![no_std]
+#![doc = include_str!("../README.md")]
+#![allow(clippy::upper_case_acronyms)]
```
Comment (tarcieri) [inline aead-stream/src/lib.rs:3]: Might be good to fix this in a separate PR

File: Cargo.lock [hunk 5]
```diff
@@ -308,9 +315,9 @@ checksum = "6fe2267d4ed49bc07b63801559be28c718ea06c4738b7a03c94df7386d2cde46"
 
 [[package]]
 name = "hybrid-array"
-version = "0.2.0-rc.9"
+version = "0.2.0-rc.10"
```
Comment (tarcieri) [inline Cargo.lock:318]: Might want to avoid this bump for now since it changes MSRV:

```suggestion
version = "0.2.0-rc.9"
```

Comment (newpavlov) [reply Cargo.lock:318]: Huh, I thought `hybrid-array` was pinned by upstream.
